app/Http/Controllers/UserController.php

```diff
@@ -67,4 +68,16 @@ public function updateUser(updateUserRequest $request, User $user)
     {
         $this->userService->updateUser($request->all(), $user);
     }
+
+    public function getUserEmails()
+    {
+        return $this->userService->getUserEmails();
+    }
+
+    public function getUserData(Request $request, User $user)
+    {
+        if ($this->authorize('view', $user)) {
```

Comment:
1) use $user->can instead of authorize method.
2) move this check to the custom request class, as we did in the update task
```
public function authorize()
{
     return $this->user()->can('update', $this->route('user'));
}
```